app/src/main/java/com/framgia/soundclound/util/BindingUtil.java

```diff
@@ -61,9 +76,57 @@ public static void setUpColorFilterHome(final TabLayout tabLayout, final int sel
         }
     }
 
+
     @BindingAdapter({"recyclerAdapter"})
     public static void setAdapter(RecyclerView recyclerView, RecyclerView.Adapter adapter) {
         recyclerView.setAdapter(adapter);
     }
 
+    public static List<Track> getToJson(String jsonTrack) {
+        Gson gson = new Gson();
+        Type type = new TypeToken<List<Track>>() {
+        }.getType();
+        return gson.fromJson(jsonTrack, type);
+    }
+
+    public static String getToList(List listObj) {
+        if (listObj == null) {
+            return null;
+        }
+        Gson gson = new Gson();
+        return gson.toJson(listObj);
+
+    }
+
+    @BindingAdapter("playlist_adapter")
+    public static void setPlaylistAdapter(RecyclerView view, RecyclerView.Adapter adapter) {
+        view.setAdapter(adapter);
+        view.setItemAnimator(new DefaultItemAnimator());
+        view.addItemDecoration(new DividerItemDecoration(view.getContext(), LinearLayoutManager.VERTICAL));
+    }
+
+    @BindingAdapter("settext")
+    public static void setNumberSong(TextView textView, int numberSong) {
+        textView.setText(String.valueOf(numberSong) + " "
```

Comment:
WARNING: [UselessStringValueOf] No need to call String.valueOf to append to a string.
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#UselessStringValueOf